Fixes an issue where TapRegion would consume taps regardless of navigation state

PrimaelQuemerais opened the pull request:
Fixes #169061
Expands on #155297

This PR fixes the issue where TapRegion with consumeOutsideTaps enabled would consume taps even after navigating to a different page.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

PrimaelQuemerais committed:
Fix TapRegion to conditionally consume outside taps based on current state

PrimaelQuemerais committed:
Merge branch 'flutter:master' into tap_region

justinmc commented on packages/flutter/lib/src/widgets/tap_region.dart:
```diff
@@ -480,7 +480,7 @@ class TapRegion extends SingleChildRenderObjectWidget {
       registry: TapRegionRegistry.maybeOf(context),
       enabled: enabled,
-      consumeOutsideTaps: consumeOutsideTaps,
+      consumeOutsideTaps: isCurrent && consumeOutsideTaps,
```
@gspencergoog Do you know if this is a reasonable solution to TapRegion listening to taps outside of its route?

justinmc reviewed (commented):
@PrimaelQuemerais Can you add a test for this?

PrimaelQuemerais committed:
test(tap_region): add regression test for consumeOutsideTaps after navigation

This test ensures that TapRegion with consumeOutsideTaps does not consume taps after navigating to a new page.

PrimaelQuemerais committed:
fix(tap_region_test): add missing context parameter to Navigator.push.

PrimaelQuemerais committed:
fix: Undefined name 'context'

PrimaelQuemerais committed:
style(tap_region_test): fix formatting

PrimaelQuemerais committed:
Merge branch 'master' into tap_region

justinmc commented on packages/flutter/test/widgets/tap_region_test.dart:
```diff
@@ -1177,4 +1177,69 @@ void main() {
     expect(count2, 1); // tapRegion2 should not respond anymore.
   });
+
+  // Regression test for the consumeOutsideTaps issue when navigating between pages
```
Just leave the GitHub URL: "Regression test for https://github.com/flutter/flutter/issues/169061."

justinmc commented on packages/flutter/test/widgets/tap_region_test.dart:
```diff
@@ -1177,4 +1177,69 @@ void main() {
+
+    await tester.pumpAndSettle();
+
+    // Navigate to the second page
```
Nit: Period at the end of the sentence here and below.

justinmc commented on packages/flutter/test/widgets/tap_region_test.dart:
```diff
@@ -1177,4 +1177,69 @@ void main() {
+      true,
+      reason: 'Button tap was not consumed by a TapRegion on a non-current route',
+    );
+  });
```
Confirmed that this test fails on master :+1:

justinmc reviewed (approved):
LGTM as this is along the same lines as https://github.com/flutter/flutter/pull/155297. FYI @chunhtai @Mairramer .

chunhtai reviewed (approved):
LGTM
